src/components/app/app.scss

```diff
@@ -3,23 +3,30 @@
 //--- Reusable site-wide styles --//
 
 .pipeline-icon {
+  opacity: 0.7;
+  transition: opacity 300ms ease;
+
   .kui-theme--light & {
-    fill: #333;
+    fill: $color-dark;
   }
 
   .kui-theme--dark & {
-    fill: #fff;
+    fill: $color-light;
+  }
+
+  .pipeline-icon-toolbar__button:hover & {
+    opacity: 1;
   }
+}
 
-  .pipeline-icon--stroke & {
-    .kui-theme--light & {
-      fill: none;
-      stroke: #333;
-    }
+.pipeline-icon.pipeline-icon--stroke {
```

Comment:
Could this not just be `.pipeline-icon--stroke {`? Why does it need the additional class? The whole point is to avoid needing to compound the classes in order to avoid the increased specificity